presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -386,6 +412,7 @@ private void invokeNoMoreSplitsIfNecessary()
                 schema,
                 partitionKeys,
                 effectivePredicate,
+                partitionMatchSupplier,
```

Comment:
Passing this to InternalHiveSplitFactory seems pointless. If the partition doesn't satisfy the criteria, the block above would've returned `COMPLETED_FUTURE` and the code will never get here. I.e., the only partitionMatchSupplier that ever gets passed to that class is one that returns true.

Reply:
IIRC dynamic filter might not be known yet when partitions are enumerated, but might be known when splits are enumerated lazily.

Reply:
Ah, I see. The `partitionMatchSupplier` might return true the first time it's called and later change to false. That's not at all obvious from the code.

Reply:
I could add a comment to explain this part, do you see any scope for code changes to make the intent of the change more clear ?